rpcs3/Emu/Cell/SPUASMJITRecompiler.cpp

```diff
@@ -121,13 +116,13 @@ void spu_recompiler::compile(spu_function_t& f)
 	compiler.alloc(vec_vars[5], asmjit::x86::xmm5);
 
 	// Initialize labels
-	std::vector<Label> pos_labels{ 0x10000 };
-	this->labels = pos_labels.data();
+	this->labels = std::unique_ptr<Label[]>(reinterpret_cast<Label*>(new u8[0x10000 * sizeof(Label)]()));
```

Comment:
Why not just `new Label[0x10000]()`?

Reply:
As the project has link time optimizations turned off, this would call the constructor 0x10000 times, which is really slow - Now its near instant.

Reply:
i assume to avoid invoking the constructor

Reply:
@TGEnigma when this code is wrong, since `unique_ptr` would call the destructor of the Labels which are technically not there.

Reply:
@danilaml True - How would you suggest solving it, considering the build settings? Calling the constructor 0x10000 for each function compilation is extremely slow

Reply:
Could make a pod wrapper type for label that contains an initialized boolean, would have to set the memory to zero however.